### PR TITLE
Added additional arguments to GenomicRDD.pipe()

### PR DESCRIPTION
**Pull request by gunjanbaid**
Use case: when input RDD partitioning needs to be preserved, in order for the piped command to work correctly. 

This is needed for piping out to MACS2, which needs to process all reads from a particular chromosome together. In addition, when there is not enough data for a particular chromosome, MACS2 needs to process multiple chromosomes together.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala [hunk 1]
```diff
@@ -484,7 +486,10 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
     val bins = GenomeBins(totalLength / rdd.partitions.size, seqLengths)
 
     // if the input rdd is mapped, then we need to repartition
-    val partitionedRdd = if (sequences.records.size > 0) {
+    val partitionedRdd = if (sequences.records.size == 0 ||
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: Prefer `sequences.records.isEmpty`

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala [hunk 2]
```diff
@@ -484,7 +486,10 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
...
+      repartitionInput == false) {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: Prefer `!repartitionInput`

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala [hunk 3]
```diff
@@ -561,7 +564,8 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
 
     // if the original rdd was aligned and the final rdd is aligned, then we must filter
     if (newRdd.sequences.isEmpty ||
-      sequences.isEmpty) {
+      sequences.isEmpty ||
+      filterOutput == false) {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala]: Prefer `!filterOutput`.

**Comment by gunjanbaid** [comment]
@fnothaft Did I miss something? [This page](https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/2423/) shows 0 failures, not sure why the above comment reports a failure.

**Comment by fnothaft** [comment]
@gunjanbaid it looks like you need to run `./scripts/format-sources`

**Comment by gunjanbaid** [comment]
@fnothaft I ran `./scripts/format-sources`, pushed updated formatting, and ran through the steps listed in [CONTRIBUTING.md](https://github.com/bigdatagenomics/adam/blob/master/CONTRIBUTING.md). Not sure why Jenkins is still failing. If there's a particular page in the [test results](https://github.com/bigdatagenomics/adam/blob/master/CONTRIBUTING.md) I am missing and should be looking at, please let me know.

**Comment by heuermh** [comment]
The Jenkins error message might be misleading, it is complaining about pom files but I believe the issue is still formatting with the file `GenomicRDD.scala`
```
++ git status --porcelain
+ test -n ' M adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala'
```
Personally I don't run `./scripts/format-sources` separately; if you run the maven build from the command line it formats the source code for you.

**Comment by gunjanbaid** [comment]
@heuermh I ran the maven build again, there were no changes to the formatting in `GenomicRDD.scala`. However, there were changes to four different POM files, though I'm not totally sure why. I committed those changes to see if that fixes the build error. 

I can add the updates to the Python/R API as well. I'll put this PR on hold until those changes have been added in.

File: adam-apis/pom.xml [hunk 4]
```diff
@@ -33,7 +33,7 @@
               As explained here: http://stackoverflow.com/questions/1660441/java-flag-to-enable-extended-serialization-debugging-info
               The second option allows us better debugging for serialization-based errors.
           -->
-          <argLine>-Xmx1024m -Dsun.io.serialization.extendedDebugInfo=true</argLine>
+          <argLine>-Xmx1024m -Dsun.io.serialization.extendedDebugInfo=true -Djava.io.tmpdir=/tmp/adamTestMvnNTasIKA</argLine>
```
Comment (akmorrow13) [inline adam-apis/pom.xml]: remove these

**Comment by heuermh** [comment]
Closing as WontFix.  Feel free to create a new PR after rebasing against git head.

**Pull request closed without merging**